### ISSUE TITLE
Make rdkafka-ruby Ractor safe

### ISSUE THREAD

**Issue by mensfeld** [karafka/rdkafka-ruby#511]

**Comment by larskanis** [comment]
I added Ractor support to ffi-1.16 one year ago and now was just looking for projects which use this feature. Anyways there is a wiki page for FFI in conjunction with Ractors: https://github.com/ffi/ffi/wiki/Ractors Would be interested in some adoption.

**Comment by mensfeld** [comment]
Hey @larskanis thanks. I know this link. My main issue with ractors atm is their limited scope of operations. Any benchmarks that I did that would involve CPU intense work (like data deserialization) eats up more resources and is more expensive in ractors than without, not to mention with forks :(

**Comment by larskanis** [comment]
Yes that's the problem. Ractor was made to improve the performance but it has so many performance issues. Among many others.

Nevertheless it's an exciting technology and I wanted ffi as an important core gem to support it properly.

**Comment by mensfeld** [comment]
@larskanis I fully agree and that is one of the reasons why I created this card in the first place :)
